{
    "url": "https://api.github.com/repos/rapidsai/cudf/issues/6810",
    "repository_url": "https://api.github.com/repos/rapidsai/cudf",
    "labels_url": "https://api.github.com/repos/rapidsai/cudf/issues/6810/labels{/name}",
    "comments_url": "https://api.github.com/repos/rapidsai/cudf/issues/6810/comments",
    "events_url": "https://api.github.com/repos/rapidsai/cudf/issues/6810/events",
    "html_url": "https://github.com/rapidsai/cudf/issues/6810",
    "id": 746926273,
    "node_id": "MDU6SXNzdWU3NDY5MjYyNzM=",
    "number": 6810,
    "title": "[BUG]Groupby agg with np.nan in aggregating column leads to incorrect results ",
    "user": {
        "login": "VibhuJawa",
        "id": 4837571,
        "node_id": "MDQ6VXNlcjQ4Mzc1NzE=",
        "avatar_url": "https://avatars.githubusercontent.com/u/4837571?v=4",
        "gravatar_id": "",
        "url": "https://api.github.com/users/VibhuJawa",
        "html_url": "https://github.com/VibhuJawa",
        "followers_url": "https://api.github.com/users/VibhuJawa/followers",
        "following_url": "https://api.github.com/users/VibhuJawa/following{/other_user}",
        "gists_url": "https://api.github.com/users/VibhuJawa/gists{/gist_id}",
        "starred_url": "https://api.github.com/users/VibhuJawa/starred{/owner}{/repo}",
        "subscriptions_url": "https://api.github.com/users/VibhuJawa/subscriptions",
        "organizations_url": "https://api.github.com/users/VibhuJawa/orgs",
        "repos_url": "https://api.github.com/users/VibhuJawa/repos",
        "events_url": "https://api.github.com/users/VibhuJawa/events{/privacy}",
        "received_events_url": "https://api.github.com/users/VibhuJawa/received_events",
        "type": "User",
        "site_admin": false
    },
    "labels": [
        {
            "id": 599626559,
            "node_id": "MDU6TGFiZWw1OTk2MjY1NTk=",
            "url": "https://api.github.com/repos/rapidsai/cudf/labels/bug",
            "name": "bug",
            "color": "d73a4a",
            "default": true,
            "description": "Something isn't working"
        },
        {
            "id": 1139741213,
            "node_id": "MDU6TGFiZWwxMTM5NzQxMjEz",
            "url": "https://api.github.com/repos/rapidsai/cudf/labels/Python",
            "name": "Python",
            "color": "1d76db",
            "default": false,
            "description": "Affects Python cuDF API."
        }
    ],
    "state": "open",
    "locked": false,
    "assignee": null,
    "assignees": [],
    "milestone": {
        "url": "https://api.github.com/repos/rapidsai/cudf/milestones/5",
        "html_url": "https://github.com/rapidsai/cudf/milestone/5",
        "labels_url": "https://api.github.com/repos/rapidsai/cudf/milestones/5/labels",
        "id": 6955120,
        "node_id": "MI_kwDOBWUGps4AaiBw",
        "number": 5,
        "title": "Pandas API Alignment and Coverage",
        "description": "",
        "creator": {
            "login": "beckernick",
            "id": 8457388,
            "node_id": "MDQ6VXNlcjg0NTczODg=",
            "avatar_url": "https://avatars.githubusercontent.com/u/8457388?v=4",
            "gravatar_id": "",
            "url": "https://api.github.com/users/beckernick",
            "html_url": "https://github.com/beckernick",
            "followers_url": "https://api.github.com/users/beckernick/followers",
            "following_url": "https://api.github.com/users/beckernick/following{/other_user}",
            "gists_url": "https://api.github.com/users/beckernick/gists{/gist_id}",
            "starred_url": "https://api.github.com/users/beckernick/starred{/owner}{/repo}",
            "subscriptions_url": "https://api.github.com/users/beckernick/subscriptions",
            "organizations_url": "https://api.github.com/users/beckernick/orgs",
            "repos_url": "https://api.github.com/users/beckernick/repos",
            "events_url": "https://api.github.com/users/beckernick/events{/privacy}",
            "received_events_url": "https://api.github.com/users/beckernick/received_events",
            "type": "User",
            "site_admin": false
        },
        "open_issues": 114,
        "closed_issues": 124,
        "state": "open",
        "created_at": "2021-07-16T17:49:59Z",
        "updated_at": "2024-06-03T17:45:11Z",
        "due_on": null,
        "closed_at": null
    },
    "comments": 6,
    "created_at": "2020-11-19T21:09:51Z",
    "updated_at": "2024-05-30T00:36:19Z",
    "closed_at": null,
    "author_association": "MEMBER",
    "active_lock_reason": null,
    "body": "**Describe the bug**\r\n\r\nIf we have nulls in the column to run aggregate on we, seem to get incorrect results if there are `np.nan` in the column we are aggregating on.    \r\n\r\n\r\n**Steps/Code to reproduce bug**\r\n```python\r\n\r\n>>> import numpy as np\r\n>>> import cudf\r\n>>> df = cudf.DataFrame({'key':[0,0,0],'val':[None,1.1,None]})\r\n>>> df['val']=df.val.fillna(np.nan)\r\n>>> df.groupby(by='key').val.sum()\r\nkey\r\n0   NaN\r\nName: val, dtype: float64\r\n```\r\n\r\n**Expected behavior**\r\n\r\n```python\r\n>>> df.to_pandas().groupby(by='key').val.sum()\r\nkey\r\n0    1.1\r\nName: val, dtype: float64\r\n```\r\n\r\n**Environment details**\r\n```python\r\ncudf                      0.17.0a201119   cuda_10.2_py37_g1a80df96c4_285    rapidsai-nightly\r\nlibcudf                   0.17.0a201119   cuda10.2_g1a80df96c4_285    rapidsai-nightly\r\n```\r\n\r\n**Additional context**\r\nAs a workaround below does the trick. \r\n\r\n```python\r\ndf.nans_to_nulls().groupby(by='key').agg({'val':['sum']})\r\n```\r\n\r\nCC: @beckernick ",
    "closed_by": null,
    "reactions": {
        "url": "https://api.github.com/repos/rapidsai/cudf/issues/6810/reactions",
        "total_count": 0,
        "+1": 0,
        "-1": 0,
        "laugh": 0,
        "hooray": 0,
        "confused": 0,
        "heart": 0,
        "rocket": 0,
        "eyes": 0
    },
    "timeline_url": "https://api.github.com/repos/rapidsai/cudf/issues/6810/timeline",
    "performed_via_github_app": null,
    "state_reason": null
}